components/nimbus/src/lib.rs

```diff
@@ -402,14 +415,24 @@ pub struct Branch {
     pub slug: String,
     pub ratio: i32,
     pub feature: Option<FeatureConfig>,
+    // We skip serializing if the `features` is none
+    // to keep backward compatibility. if we do not skip it,
+    // new experiments that only have one feature will end up
+    // serializing to a `features: Null` which does not match what
+    // old experiments were serialized to when they were persisted
+    // since they do not include this field at all.
+    #[serde(skip_serializing_if = "Option::is_none")]
```

Comment:
Nice!

Reply:
Can we add this to `feature` as well?

Reply:
That would bring us to the same question I have in https://github.com/mozilla/application-services/pull/4452/files#r709390143

Do we care that after deserializing and serializing an old json, it looks different? If not, we can do that, if we do, we might have to keep serializing the `feature` until we do a migration 😞

Reply:
I think the current roundtripping tests get this backwards. 

If we're doing a test comparing the serializations:

i.e. `in json_string` --> `Experiment` --> `out json_string`; then test `in json_string` == `out json_string`.

We should be doing a test comparing the live objects themselves:

i.e. `in Experiment` --> `json_string` --> `out Experiment`; then test `in Experiment` == `out Experiment`.